### PR TITLE
Get URL and API key from Cloud/emulator-specified env vars to be easi…

### PR DESCRIPTION
**Pull request by SavagePencil**
…er for customers

### CODE REVIEW

File: structure/structure.py [hunk 1]
```diff
@@ -15,24 +15,52 @@
...
+    environment variable.
+    In Griptape Cloud, this environment variable is provided for you.
+    In Skatepark, you may optionally provide your own if you have overridden
+    the default URL and port that the emulator is listening on. Otherwise,
+    it will use the default URL and port.
+
+    Returns:
+        The URL of the host listening to the Structure.
+    """
+    return os.environ.get("GT_CLOUD_BASE_URL")
```
Comment (collindutter) [inline structure/structure.py]: The framework will do this for the user, not needed.

File: structure/structure.py [hunk 2]
```diff
@@ -15,24 +15,52 @@
...
+    When your program is running in Griptape Cloud, you will need to provide a 
+    valid Griptape API Key in the GT_CLOUD_API_KEY environment variable 
+    in order to authorize calls. 
+    You can create an API Key by visiting https://cloud.griptape.ai/keys 
+    When running in Skatepark, the API key is not needed since it isn't validating calls.
+
+    Returns:
+        The Griptape API Key to authorize calls.
+    """
+    return os.environ.get("GT_CLOUD_API_KEY")
```
Comment (collindutter) [inline structure/structure.py]: This is where the the warning should go, yes?

File: structure/structure.py (hunk 1 above)
Comment (SavagePencil) [reply structure/structure.py]: a) I wrote this before I saw your framework change
b) where's a good place to tell the user about it?

File: structure/structure.py (hunk 2 above)
Comment (SavagePencil) [reply structure/structure.py]: I thought we could put the warning in Skatepark itself, right? If you'd prefer it here, I can do it here.

Comment (collindutter) [reply structure/structure.py]: Oh I see. We can add it to Skatepark but the user may not be looking at the server logs.

File: structure/structure.py (hunk 1 above)
Comment (collindutter) [reply structure/structure.py]: b) I added the info in the `GriptapeCloudEventListenerDriver` class docstring but that still may not be enough. Maybe just a comment near the Driver instantiation?

File: structure/structure.py (hunk 2 above)
Comment (SavagePencil) [reply structure/structure.py]: I'll put a warning here.

File: structure/structure.py (hunk 1 above)
Comment (collindutter) [reply structure/structure.py]: @SavagePencil can we delete this now?

File: structure/structure.py [hunk 3]
```diff
@@ -15,24 +15,66 @@
...
 input = sys.argv[1]
 
-# We need an event driver to communicate events from our program back to Skatepark
-listener_base = os.environ.get("GRIPTAPE_SKATEPARK_URL_OVERRIDE", "http://127.0.0.1")
-listener_port = os.environ.get("GRIPTAPE_SKATEPARK_PORT_OVERRIDE", "5000")
-listener_url = f"{listener_base}:{listener_port}"
-
+# We need an event driver to communicate events from our program back to our
+# host, which could be the locally-run Skatepark or Griptape Cloud.
+# The event driver requires a URL, stored in the GT_CLOUD_BASE_URL environment variable.
```
Comment (collindutter) [inline structure/structure.py]: This reads a bit like it's _required_ for it to be in `GT_CLOUD_BASE_URL`. We read from there by default, but users can store it wherever they'd like and then manually pass it in.

File: structure/structure.py [hunk 4]
```diff
@@ -15,24 +15,66 @@
...
+# In Griptape Cloud, this environment variable is provided for you.
```
Comment (collindutter) [inline structure/structure.py]: This environment variable is provided in Skatepark too. The behavior is identical across Cloud/Skatepark, the value is the only thing that will change.

**Review by collindutter — CHANGES_REQUESTED** [review]

File: structure/structure.py (hunk 3 above)
Comment (SavagePencil) [reply structure/structure.py]: ah yeah, poor choice of tone. Will update.

File: structure/structure.py (hunk 4 above)
Comment (SavagePencil) [reply structure/structure.py]: Yeah, I was trying to call out each runtime environment separately. let me do some ASTERISK MAGIC.

**Comment by collindutter** [comment]
Also need to resolve conflicts pls

File: structure/structure.py [hunk 5]
```diff
@@ -15,24 +15,52 @@
...
 
-# We need an event driver to communicate events from our program back to Skatepark
-listener_base = os.environ.get("GRIPTAPE_SKATEPARK_URL_OVERRIDE", "http://127.0.0.1")
-listener_port = os.environ.get("GRIPTAPE_SKATEPARK_PORT_OVERRIDE", "5000")
-listener_url = f"{listener_base}:{listener_port}"
-
+# We need an event driver to communicate events from our program back to our
+# host, which could be the locally-run Skatepark or Griptape Cloud.
+# The event driver requires a URL to the host.
+# * In Griptape Cloud, this environment variable is automatically provided to you in GT_CLOUD_BASE_URL.
```
Comment (collindutter) [inline structure/structure.py]: I think we want something like:

> When running in Skatepark or Griptape Cloud, this value is automatically provided to you in the environment variable `GT_CLOUD_BASE_URL`. You can also choose to override this value by setting `GriptapeCloudEventListenerDriver.base_url`.

**Review by collindutter — APPROVED** [review]

**Review by shhlife — APPROVED** [review]
LGTM! :)

**Comment by SavagePencil** [comment]
Updated to simplify local dev -> local emulator -> deployed to Griptape Cloud

**Pull request merged**